packages/frontend/src/server/features/scaling/costs/get-costs-chart.ts

```diff
@@ -46,20 +46,26 @@ const getCachedCostsChart = cache(
     }
     const resolution = rangeToResolution(timeRange)
     const targetTimestamp = getCostsTargetTimestamp()
-    const range = getRange(timeRange, resolution, { now: targetTimestamp })
+    const [from, to] = getRange(timeRange, resolution, { now: targetTimestamp })
+
+    // one-off
+    const fromToQuery = from.add(-1, resolution === 'daily' ? 'days' : 'hours')
+
+    // to is exclusive
+    const rangeForQuery: [UnixTime, UnixTime] = [fromToQuery, to]
+
     const data = await db.aggregatedL2Cost.getByProjectsAndTimeRange(
       projects.map((p) => p.id),
-      range,
+      rangeForQuery,
     )
 
     if (data.length === 0) {
       return []
     }
 
     const summedByTimestamp = sumByTimestamp(data, resolution)
-    const [from, to] = range
     const timestamps = generateTimestamps(
-      [from, to.add(-1, resolution === 'daily' ? 'days' : 'hours')],
+      [fromToQuery, to.add(-1, resolution === 'daily' ? 'days' : 'hours')],
```

Comment:
```suggestion
      rangeForQuery,
```

Reply:
How can this be wrong? Database query convention stays the same, and without extending the range we'll be lacking data